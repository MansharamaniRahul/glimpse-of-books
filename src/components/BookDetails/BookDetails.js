import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import styles from './BookDetails.module.scss';
import Loader from 'react-loader-spinner';

class BookDetails extends Component {
	render() {
		const { BooksStore } = this.props;

		return (
			<React.Fragment>
				<div className={styles.container}>
					{BooksStore.showLoader && (
						<div className={styles.loader}>
							{' '}
							<Loader
								type="Puff"
								color="#00BFFF"
								height="100"
								width="100"
							/>
						</div>
					)}
					{BooksStore.selectedBookDesc && (
						<div>
							<div className={styles.img}>
								<img
									src={
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book.image_url
											._text
									}
								/>
							</div>

							<div className={styles.bookDescContainer}>
								<div className={styles.bookName}>
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book.title._text
									}
								</div>

								<div className={styles.bookAuthor}>
									by :
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book.authors
											.author.name._text
									}
								</div>

								<div className={styles.rating}>
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book
											.average_rating._text
									}{' '}
									&nbsp;&nbsp;&nbsp; Rating Details:{' '}
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book
											.ratings_count._cdata
									}{' '}
									ratings. &nbsp;&nbsp;&nbsp;
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book
											.text_reviews_count._cdata
									}{' '}
									reviews.
								</div>

								<div className={styles.bookDesc}>
									{
										BooksStore.selectedBookDesc
											.GoodreadsResponse.book.description
											._cdata
									}
								</div>
							</div>
						</div>
					)}
				</div>
			</React.Fragment>
		);
	}
}

export default inject('BooksStore')(observer(BookDetails));
