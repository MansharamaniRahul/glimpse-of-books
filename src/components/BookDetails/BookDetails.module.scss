.container {
    width: 85%;    
    height: 100%;    
    margin: 6%;    
        padding: 2%;    
    float: left;    
    background-color:     #f7f8f9;
        font-family: 
    Arial, Helvetica, 
    sans-serif;    
    border-radius:     30px;    
    }
    
    .img {    
    width: 10%;    
    height: 100%;    
    float: left;    
    position: relative;    
    }
    
    .bookDescContainer {    
    width: 90%;    
    height: 100%;    
    float: left;    
    position: relative;    
    }
    
    .bookName {    
    float: left;    
    font-weight:     bold;    
    font-family:     "Merriweather", "Georgia",     serif;    
    font-size: 24px;    
    line-height:     1.25;   
    }
    
.bookAuthor{
    
    float: left;
    
    clear: left;
    
    line-height: 
    18px;
    
    font-family: 
    "Lato", "Helvetica Neue", 
    Helvetica, Arial, 
    sans-serif;
    
    text-align: left;
    
    font-size: 12px;
    
    margin-top: 7px
    !important;
    
    margin-bottom: 
    7px !important;
    
    }
    
    .rating {
    
    float: left;
    
    clear: left;
    
    color: #00635d;
    
    font-family: 
    "Lato", "Helvetica Neue", 
    Helvetica, Arial, 
    sans-serif;
    
    font-size: 12px;
    
    margin-top: 7px
    !important;
    
    margin-bottom: 
    7px !important;
    
    
    }
    
    .bookDesc {
    
    float: left;
    
    clear: left;
    
    text-align: left;
    
    line-height: 
    21px;
    
    font-family: 
    "Merriweather", "Georgia", 
    serif;
    
    font-size: 14px;
    
    display: block;
    
    }

    .loader{
        width: 100%;
        height: 100%;
       
    }