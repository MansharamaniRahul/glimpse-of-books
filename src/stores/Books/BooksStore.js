import { decorate, observable, action } from 'mobx';

import makeInspectable from 'mobx-devtools-mst';

import BooksStoreServiceAPI from '../../services/BooksStoreServices/BooksStoreService';

var convert = require('xml-js');

class BooksStore {
	bookName = '';

	booksList = [];

	selectedBookDesc = '';

	errorCode = '';
	showLoader = false;

	fetchBooksListFromServer = bookName => {
		if (this.booksList.length > 0) this.booksList.length = 0;
		this.setLoader();
		BooksStoreServiceAPI.getBooksList(bookName)

			.then(
				res => {
					var jsonConvertedData = convert.xml2json(res.data, {
						compact: true,

						spaces: 4
					});

					this.setBooksList(jsonConvertedData);
					this.setLoader();
				},

				error => {
					console.log(`http error here
	${error}`);
					this.setLoader();

					this.setErrorCode(error.message);
				}
			)

			.catch(error => {
				console.log('other error here', error);
			});
	};

	setBooksList = data => {
		data = JSON.parse(data);

		if (data) this.booksList.push(data);
		else alert('No Books received');
	};

	setErrorCode = errorCode => {
		this.errorCode = errorCode;
	};

	getErrorCode = () => {
		return this.errorCode;
	};

	fetchSelectedBookDesc = id => {
		if (this.selectedBookDesc !== '') this.selectedBookDesc = '';
		this.setLoader();
		BooksStoreServiceAPI.getBookDesc(id)

			.then(
				res => {
					this.setLoader();

					var jsonConvertedData = convert.xml2json(res.data, {
						compact: true,

						spaces: 4
					});

					this.setSelectedBookDesc(jsonConvertedData);
				},

				error => {
					console.log(`http error here
    ${error}`);

					this.setErrorCode(error.message);
				}
			)

			.catch(error => {
				console.log('other error here', error);
			});
	};

	setSelectedBookDesc = data => {
		this.selectedBookDesc = JSON.parse(data);
	};

	getSelectedBookDesc = () => {
		return this.selectedBookDesc;
	};

	setLoader = () => {
		this.showLoader = !this.showLoader;
	};
}

decorate(BooksStore, {
	bookName: observable,

	selectedBookDesc: observable,

	booksList: observable,
	showLoader: observable,

	errorCode: observable,

	setBooksList: action.bound,

	setErrorCode: action.bound,

	getBooksList: action.bound,

	getErrorCode: action.bound,

	searchBook: action.bound,
	setLoader: action.bound
});

export default makeInspectable(new BooksStore());
