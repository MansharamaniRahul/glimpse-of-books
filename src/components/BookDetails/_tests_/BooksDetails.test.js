import BookDetails from "../BookDetails";
import { shallow, configure } from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
import { Provider } from "mobx-react";
import BooksStore from "../../../stores/Books/BooksStore";
configure({ adapter: new Adapter() });
const testProps = {
  id: "1",
  bookTitle: "xyz",
  imageURL:
    "https://www.google.com/imgres?imgurl=https%3A%2F%2Fcdn0.tnwcdn.com%2Fwp-content%2Fblogs.dir%2F1%2Ffiles%2F2018%2F05%2FUntitled-design-23-796x417.png&imgrefurl=https%3A%2F%2Fthenextweb.com%2Ftq%2F2018%2F05%2F22%2Ftactics-driving-business-growth%2F&docid=U0DMo3my-pUdxM&tbnid=9nrfL9UNpYJY2M%3A&vet=10ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA..i&w=796&h=417&bih=821&biw=1440&q=grow&ved=0ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA&iact=mrc&uact=8",

  imageURLSmall:
    "https://www.google.com/imgres?imgurl=https%3A%2F%2Fcdn0.tnwcdn.com%2Fwp-content%2Fblogs.dir%2F1%2Ffiles%2F2018%2F05%2FUntitled-design-23-796x417.png&imgrefurl=https%3A%2F%2Fthenextweb.com%2Ftq%2F2018%2F05%2F22%2Ftactics-driving-business-growth%2F&docid=U0DMo3my-pUdxM&tbnid=9nrfL9UNpYJY2M%3A&vet=10ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA..i&w=796&h=417&bih=821&biw=1440&q=grow&ved=0ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA&iact=mrc&uact=8",

  rating: "3",

  author: "abc",

  pubYear: "1990",

  pubMonth: "2",

  pubDate: "2"
};

describe("<BookDetails />", () => {
  it("matches the snapshot", () => {
    const tree = shallow(
      <Provider Bookstore={BooksStore}>
        <BookDetails {...testProps} />
      </Provider>
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
