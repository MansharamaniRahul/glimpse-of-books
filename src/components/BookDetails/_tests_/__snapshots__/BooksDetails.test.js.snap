// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetails /> matches the snapshot 1`] = `
<inject-BookDetails-with-BooksStore
  author="abc"
  bookTitle="xyz"
  id="1"
  imageURL="https://www.google.com/imgres?imgurl=https%3A%2F%2Fcdn0.tnwcdn.com%2Fwp-content%2Fblogs.dir%2F1%2Ffiles%2F2018%2F05%2FUntitled-design-23-796x417.png&imgrefurl=https%3A%2F%2Fthenextweb.com%2Ftq%2F2018%2F05%2F22%2Ftactics-driving-business-growth%2F&docid=U0DMo3my-pUdxM&tbnid=9nrfL9UNpYJY2M%3A&vet=10ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA..i&w=796&h=417&bih=821&biw=1440&q=grow&ved=0ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA&iact=mrc&uact=8"
  imageURLSmall="https://www.google.com/imgres?imgurl=https%3A%2F%2Fcdn0.tnwcdn.com%2Fwp-content%2Fblogs.dir%2F1%2Ffiles%2F2018%2F05%2FUntitled-design-23-796x417.png&imgrefurl=https%3A%2F%2Fthenextweb.com%2Ftq%2F2018%2F05%2F22%2Ftactics-driving-business-growth%2F&docid=U0DMo3my-pUdxM&tbnid=9nrfL9UNpYJY2M%3A&vet=10ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA..i&w=796&h=417&bih=821&biw=1440&q=grow&ved=0ahUKEwi6zIuX97PhAhWEknAKHal1CDUQMwhcKAAwAA&iact=mrc&uact=8"
  pubDate="2"
  pubMonth="2"
  pubYear="1990"
  rating="3"
/>
`;
